### DISCUSSION TITLE
Who can use Codespace prebuilds for a repo or template repo?

### DISCUSSION THREAD

**Discussion by aaronsteers** [community/community#39287] [Codespaces] [answered]
### Select Topic Area

Question

### Body

In reading the docs, I could not tell whether [prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds) would be available to external users from an organization, or only to the owners/members of the repo or organization. Can you help us clarify this point?

And if they are available to external users, would they also be available to speed up Codespaces from Template repos?

When inviting potential users to try out our product, we'd love for them to get the fastest possible startup experience when using Codespaces. 😄

**Comment by erikaellr** [comment, marked as answer]
Hi @aaronsteers, we're thrilled you're using Codespaces!

Yes, a repo's prebuilds are available to any users who have access to a repo, including any organization members, invited external collaborators, and even the public if the repo is public.

Yes, prebuilds can be used with template repos - the configuration is the same, and any user who clicks "Use this template" and chooses the "Open in a codespace" option will have their codespace start time benefit from use of the template repo's prebuild.

**Reply by aaronsteers** [reply]
@erikaellr - fantastic! We're excited to start using this "in the wild" 😁 - thanks so much!

**Comment by njelich** [comment]
@erikaellr 

Is it possible to share the prebuild with those that select **Use this template > Create a new repository**? 

Are there any plans to add that as an option?

**Reply by jeffbax** [reply]
Would also really love to see this. At the moment starting a new repo (for something like a technical interview) will drop access to the prebuild that is still the starting point we want to use over and over again.

The only way you can use the prebuild currently seems to be
1. Open codespace from template
2. Push to a private repo (at least I can't seem to choose the Org as a destination)
3. Re-transfer that repo to the organization

Otherwise you have to
1. Create repo from template
2. Start build from scratch
